torcheeg/models/gan/eegfusenet.py

```diff
@@ -0,0 +1,231 @@
+import torch.nn as nn
+import torch.nn.functional as F
+import torch
+
+class EEGfuseNet(nn.Module):
+    r'''
+    EEGFuseNet: A hybrid unsupervised network which can fuse high-dimensional EEG to obtain deep feature characterization and generate similar signals. For more details, please refer to the following information.
+
+    - Paper: Z. Liang, R. Zhou, L. Zhang, L. Li, G. Huang, Z. Zhang, and S. Ishii, EEGFuseNet: Hybrid Unsupervised Deep Feature Characterization and Fusion for High-Dimensional EEG With an #Application to Emotion Recognition, IEEE Transactions on Neural Systems and Rehabilitation Engineering, 29, pp. 1913-1925, 2021.
+    - URL: https://github.com/KAZABANA/EEGfusenet
+
+    .. code-block:: python
+        g_model=EEGfuseNet(in_channels=20,hidden_dim=16,n_layers=1,n_filters=1,chunk_size=128)
+        X = torch.rand(128,20,128)
+        fake_X,deep_feature=g_model(X)
+
+    Args:
+        in_channels (int): The input feature dimension. (default: :obj:`128`)
+        hidden_dim  (int): Hidden dim of BI-GRU. (default: :obj:`16`)
+        n_layers (int): The number of layers of BI-GRU. (default: :obj:`1`)
+        n_filters (int): The number of filters of CNN based encoder. (default: :obj:`1`)
+        chunk_size (int): Number of data points included in each EEG chunk.the size of the input EEG signal is( batch size × Channel × Time) (default: :obj:`128`)
+    '''
+    def __init__(self,
+                    in_channels: int = 32,
+                    hidden_dim: int = 16,
+                    n_layers: int = 1,
+                    n_filters: int = 1,
+                    chunk_size:int = 384 ):  
+        super(EEGfuseNet, self).__init__()
+        
+        
+        self.n_filters = n_filters
+        self.hidden_dim =  hidden_dim
+        self.length = chunk_size/32
+
+        self.conv1 = nn.Conv2d(1, 16*n_filters, (1, int(chunk_size/2+1)),stride = 1, padding = (0,int(chunk_size/4)))
```

Comment:
Please make in_channels configurable, with a default of 1 (used for multiple filter banks).